gordon/main.py

```diff
@@ -195,7 +195,7 @@ def run(config_root):
 
     route_config = config.get('core', {}).get('route', {})
     msg_router = _setup_router(
-        route_config, message_handlers, metrics, **channels)
+        route_config, message_handlers, **plugin_kwargs)
```

Comment:
In plugin_loader.py, we're editing this dict value. Adding a `plugin_kwargs.copy()` adds a bit of overhead, although negligible. I figured since we're setting plugin kwargs to plugins and not separating out the `metrics` object, that we could keep the same convention here. Open to other ideas though!

Reply:
Ah, good catch! I didn't know plugin_kwargs was used outside of the plugin loader, but I guess that's just because of the different names. There's some argument to be made for not editing shared resources inside of a function, since it's more opaque. *If* you wanted to do it a different way, you could return `plugin_kwargs` from `load_plugins` instead of `metrics`.  That might make it more obvious that it has different values than the dict that went into the function.

Reply:
@descent3me I like that approach - updated!